turbo.json

```diff
@@ -29,6 +29,14 @@
       "dependsOn": ["^build"],
       "outputs": []
     },
+    "block-template-custom-element#codegen": {
+      "dependsOn": ["^build"],
+      "outputs": ["src/types.gen.ts"]
+    },
+    "block-template-react#codegen": {
+      "dependsOn": ["^build"],
+      "outputs": ["src/types.gen.ts"]
+    },
```

Comment:
The codegen here relies on `@blockprotocol/graph` being built, because it uses the type generator from it.

Reply:
It would be nice to see that explicitly defined.. can we not represent that in the syntax easily? Or is it (specifically needing `@blockprotocol/graph`) just implied by the `package.json` magic

Reply:
Not here, no – `dependsOn` allows you to define that a job depends on either a job within the same package (e.g. `build`) or depends on a job being completed in all its dependencies, as detected in package.json (e.g. `^build`). You cannot specify that it depends on a particular package's job being completed.

Reply:
[[docs](https://turbo.build/repo/docs/reference/configuration#dependson)]

Reply:
I wonder if nx lets you do something more fine-grained

Reply:
From the [docs](https://nx.dev/reference/project-configuration#dependson) at least, it looks like it's the exact same syntax (albeit with a more verbose syntax that still doesn't have any indication / example of being able to go below the level of `self` or `deps`)